s3/aide.go

```diff
@@ -30,11 +30,23 @@ type Service struct {
 // ACL is bucket-owner-full-control by default, but can be changed with SetACL.
 // SSE is AES256 by default, but can be changed with SetSSE.
 func New(name string, region, roleARN *string) *Service {
+	service := new(name)
+	service.svc = s3.New(aidews.Session(region, roleARN))
+	return service
+}
+
+// NewWithConfig return a pointer to a new Service using a provided aws.Config object
+func NewWithConfig(name string, cfg aws.Config, roleARN *string) *Service {
+	service := new(name)
+	service.svc = s3.New(aidews.SessionWithConfig(cfg, roleARN))
+	return service
+}
+
+func new(name string) *Service {
```

Comment:
I dig the change, but I have a slight variation to suggest. Having seen it, I suggest the functions: `New`, `NewWithConfig`, and `newWithSvc`. I think it improves the shape a bit, and avoids using `new` which conflates the meaning of the built-in keyword. Something like this.

``` go
func New(name string, region, roleARN *string) *Service {
	return newWithSvc(name, s3.New(aidews.Session(region, roleARN)))
}

func NewWithConfig(name string, cfg aws.Config, roleARN *string) *Service {
	return newWithSvc(name, s3.New(aidews.SessionWithConfig(cfg, roleARN)))
}

func newWithSvc(name string, svc s3iface.S3API) *Service {
	return &Service{
		acl:  aws.String("bucket-owner-full-control"),
		name: name,
		sse:  aws.String("AES256"),
		svc:  svc,
	}
}